packages/deployment/test/test_lambda.js

```diff
@@ -87,7 +87,42 @@ test.serial('zipLambda: works for lambda using message adapter', async (t) => {
 });
 
 test.serial(
-  `zipLambda: for lambda using message adapter, no new file is generated 
+  `zipLambda: given an invalid zip file generated from a previous run,
+  a new valid lambda file is generated`,
+  async (t) => {
+    t.context.lambda.useMessageAdapter = true;
+    const lambdaLocalOrigin = t.context.lambda.local;
+    const lambdaRemoteOrigin = t.context.lambda.remote;
+
+    // put an empty lambda zip file there as the result of the previous run
+    const existingLambdaLocal = path.join(
+      path.dirname(t.context.lambda.local),
+      `${Lambda.messageAdapterZipFileHash}-${path.basename(t.context.lambda.local)}`
+    );
+
+    fs.writeFileSync(existingLambdaLocal, 'hello');
```

Comment:
Instead of `fs.writeFileSync`, look at using the fs-extra module and `await fs.writeFile`.

Reply:
@yjpa7145 out of curiosity, what is the benefit of doing an `await fs.writeFile` over `fs.writeFileSync` when on the next line the result of `writeFile` is being referenced?

Reply:
oh, I think I see. If the next line is also async, then everything makes sense.

Reply:
@ifestus If you use the Sync call, the entire process stops until that writeFile operation finishes.  That means that any other background operations (any asynchronous code running in the process) will hang until this write operation finishes.  If the async version is used then those other asynchronous operations can continue while this write is being performed.